core/src/main/java/org/elasticsearch/action/index/TransportIndexAction.java

```diff
@@ -88,6 +89,9 @@ protected void doExecute(final IndexRequest request, final ActionListener<IndexR
         // if we don't have a master, we don't have metadata, that's fine, let it find a master using create index API
         ClusterState state = clusterService.state();
         if (autoCreateIndex.shouldAutoCreate(request.index(), state)) {
+            if (!settings.getAsBoolean("index.mapper.dynamic", true)) {
```

Comment:
Also, I think that we should factor the default value out into a constant as well, and reuse it here and in `MapperService`.